lib/astarte/core/generators/value.ex

```diff
@@ -0,0 +1,114 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Generators.Value do
+  @moduledoc """
+  This module provides generators for any Value type.
+  """
+  use ExUnitProperties
+
+  alias Astarte.Core.Interface
+  alias Astarte.Core.Mapping
+
+  alias Astarte.Common.Generators.DateTime, as: DateTimeGenerator
+  alias Astarte.Common.Generators.Timestamp, as: TimestampGenerator
+  alias Astarte.Core.Generators.Mapping, as: MappingGenerator
+
+  @doc """
+  Generates a valid value based on interface
+  """
+  @spec value(Interface.t()) :: StreamData.t(map())
+  def value(%Interface{} = interface) when not is_struct(interface, StreamData) do
+    interface |> constant() |> value()
+  end
+
+  @spec value(StreamData.t(Interface.t())) :: StreamData.t(map())
+  def value(gen) do
+    gen all %Interface{
+              mappings: mappings,
+              aggregation: aggregation
+            } <- gen,
+            %Mapping{
+              endpoint: endpoint
+            } <- member_of(mappings),
+            endpoint = interface_endpoint(aggregation, endpoint),
+            path <- endpoint_path(endpoint),
+            value <- build_value(aggregation, mappings) do
+      %{
+        path: path,
+        value: value
+      }
+    end
+  end
+
+  defp interface_endpoint(:individual, endpoint), do: endpoint
+  defp interface_endpoint(:object, endpoint), do: String.replace(endpoint, ~r"/[^/]+$", "")
+
+  defp endpoint_path(endpoint) do
+    endpoint
+    |> String.split("/")
+    |> Enum.map(&convert_token/1)
+    |> fixed_list()
+    |> map(&Enum.join(&1, "/"))
+  end
+
+  defp convert_token(token) do
+    case(Mapping.is_placeholder?(token)) do
+      true -> MappingGenerator.endpoint_segment()
+      false -> constant(token)
+    end
+  end
+
+  defp build_value(:individual, [%Mapping{value_type: value_type} | _]) do
+    value_from_type(value_type)
+  end
+
+  defp build_value(:object, [%Mapping{} | _] = mappings) do
```

Comment:
I'm not sure you need to pattern match here?

```suggestion
  defp build_value(:object, mappings) do
```